tests/test_dsl.py

```diff
@@ -320,3 +327,321 @@ def assert_cohorts_equivalent(dsl_cohort, qm_cohort):
+    assert not output.value.negated
+    assert output.value.operator == "__lt__"
+
+    assert isinstance(output.value.lhs, ValueFromRow)
+    assert output.value.lhs.column == "number"
+
+    assert isinstance(output.value.rhs, IntSeries)
+    assert isinstance(output.value.rhs.value, ValueFromRow)
+    assert output.value.rhs.value.column == "number"
+
+
+def test_intseries_le():
+    series1 = IntSeries(ValueFromRow(source=None, column="number"))
+    series2 = IntSeries(ValueFromRow(source=None, column="number"))
+
+    output = series1 <= series2
+    assert isinstance(output, BoolSeries)
+
+    # do we get the expected Comparator structure?
+    assert output.value.connector is None
+    assert not output.value.negated
+    assert output.value.operator == "__le__"
+
+    assert isinstance(output.value.lhs, ValueFromRow)
+    assert output.value.lhs.column == "number"
+
+    assert isinstance(output.value.rhs, IntSeries)
+    assert isinstance(output.value.rhs.value, ValueFromRow)
+    assert output.value.rhs.value.column == "number"
+
+
+def test_intseries_ne():
+    series1 = IntSeries(ValueFromRow(source=None, column="number"))
+    series2 = IntSeries(ValueFromRow(source=None, column="number"))
+    output = series1 != series2
+    assert isinstance(output, BoolSeries)
+
+    # do we get the expected Comparator structure?
+    assert output.value.connector is None
+    assert not output.value.negated
+    assert output.value.operator == "__ne__"
+
+    assert isinstance(output.value.lhs, ValueFromRow)
+    assert output.value.lhs.column == "number"
+
+    assert isinstance(output.value.rhs, IntSeries)
+    assert isinstance(output.value.rhs.value, ValueFromRow)
+    assert output.value.rhs.value.column == "number"
+
+
+def test_patientseries_eq():
+    output = PatientSeries(value=Value()) == PatientSeries(value=Value())
+
+    assert isinstance(output, BoolSeries)
+
+    # do we get the expected Comparator structure?
+    assert output.value.connector is None
+    assert not output.value.negated
+    assert output.value.lhs == ""
+    assert output.value.operator == "__eq__"
+
+    assert isinstance(output.value.rhs, PatientSeries)
+    assert output.value.rhs.value == ""
+
+
+def test_patientseries_is_comparator():
+    assert not PatientSeries(value=Value())._is_comparator()
+
+    assert PatientSeries(value=Comparator())._is_comparator()
+
+
+def test_patientseries_invert():
+    series = ~PatientSeries(value=Comparator())
+    assert isinstance(series.value, Comparator)
+
+    series = ~PatientSeries(value=Value())
+    assert isinstance(series.value, Comparator)
+
+
+def test_patientseries_repr():
+    events = tables.clinical_events
+    series = events.sort_by(events.date).first_for_patient().select_column(events.code)
+
+    expected = "CodeSeries(value=ValueFromRow(source=Row(source=Table(name='clinical_events'), sort_columns=('date',), descending=False), column='code'))"
+    assert repr(series) == expected
```

Comment:
What does this test?

Reply:
We build the repr from the given series of a Column, so here `events.code` is a `CodeColumn`, and should return a `CodeSeries`.  Since we then use a Series' repr when hashing I thought it was useful to test it worked as expected.  Would it be helpful to add this explanation to the test too?

Reply:
From the perspective of hashing, as long as `PatientSeries.__repr__()` is a pure function (and it'd be batshit if it wasn't, so we don't need to test for that) it doesn't really matter what it returns.

So I think we're just testing this for code coverage, which is fine.

Reply:
As we're now using `repr` for testing equality, I think we can drop this test.